text/0000-dot-profile-utility-buildpack.md

```diff
@@ -0,0 +1,129 @@
+It will:
+
+- Detect a `.profile` file in the app dir
+- Wrap the `.profile` file so that it implements the `exec.d` interface
+- Add the `exec.d` executable to the `<layer>/exec.d` directory so the launcher will apply it
+
+We propose the ID `buildpacksio/profie` for this buildpack (cf. [RFC 97](https://github.com/buildpacks/rfcs/blob/main/text/0097-official-utility-buildpacks.md#what-it-is)).
+
+## Example 1 (environment variables)
+
+Here is an example of a `.profile` script, inspired by paketo-buildpacks/node-engine:
+
+```
+memory_in_bytes="$(cat /sys/fs/cgroup/memory/memory.limit_in_bytes)"
+MEMORY_AVAILABLE="$(( $memory_in_bytes / ( 1024 * 1024 ) ))"
+export MEMORY_AVAILABLE
+```
+
+The wrapper should ensure that the `MEMORY_AVAILABLE` environment variable is set in the environment with the proper value.
+
+## Example 2 (file system side effects)
+
+With this `.profile` script:
+
+```
+echo 'hello' >> "$HOME/hello"
+```
+
+The wrapper would not need to set any environment varibales, but should maintain the side effect of creating the `$HOME/hello` file.
+
+
+# How it Works
+[how-it-works]: #how-it-works
+
+In short, the wrapper should do:
+
+```
+#!/bin/bash
+set -euo pipefail
+source .profile
+env >&3
+```
+
+We acknowledge that the `env` output will need to be properly quoted to be valid TOML.
+We will use or create a tool to handle that formatting.
+
+This will write the environment, including any variables set in `.profile`, to the [`exec.d` output TOML](https://github.com/buildpacks/spec/blob/main/buildpack.md#execd-output-toml).
+Since it also executes the `.profile` script, any side effects will happen.
+This will solve for both of the simple examples above.
+
+On Windows, for `.profile.bat` scripts, we can take the same approach of wrapping the script like:
+
+```
+call .profile.bat
+set >&%CNB_EXEC_D_HANDLE%
+```
+
+# Migration
+[migration]: #migration
+
+This buildpack is new, so has no inherent migration considerations.
+
+For the related migration concerns for removing shell functionality in general, see the [Migration Path section of RFC 93](https://github.com/buildpacks/rfcs/blob/main/text/0093-remove-shell-processes.md#migration-path).
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+This is a new component to maintain and support.
+That being said, this will enable spec and lifecycle simplifications noted in [RFC 93](https://github.com/buildpacks/rfcs/blob/main/text/0093-remove-shell-processes.md) (where this approach was first suggested).
+Also, [RFC 97](https://github.com/buildpacks/rfcs/blob/main/text/0097-official-utility-buildpacks.md) resolves to support and maintain utility buildpacks.
+
+# Alternatives
+[alternatives]: #alternatives
+
+If we do nothing, we introduce a regression in functionality, and force application developers to rework their `.profile` scripts.
+
+
+# Prior Art
+[prior-art]: #prior-art
+
+-
```

Comment:
https://devcenter.heroku.com/articles/buildpack-api#profile-d-scripts

Reply:
Thank you. Started with this and fleshed out some prior art.